### PR TITLE
feat(rpc): add EthSigner trait

### PR DESCRIPTION
**Pull request by mattsse**
Adds `EthSigner` abstraction. 

This will be used for all sign related RPC calls and for signing `eth_sendTransaction` requests.

### CODE REVIEW

File: crates/net/rpc/src/eth/api/mod.rs [hunk 1]
```diff
@@ -48,7 +50,7 @@ where
 {
     /// Creates a new, shareable instance.
     pub fn new(client: Arc<Client>, pool: Pool, network: Network) -> Self {
-        let inner = EthApiInner { client, pool, network };
+        let inner = EthApiInner { client, pool, network, signers: vec![] };
```
Comment (rkrasiuk) [inline crates/net/rpc/src/eth/api/mod.rs:53]:
```suggestion
        let inner = EthApiInner { client, pool, network, signers: Vec::default() };
```

File: crates/net/rpc/src/eth/signer.rs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+use std::collections::HashMap;
+
+/// An Ethereum Signer used via RPC.
+#[async_trait::async_trait]
+pub(crate) trait EthSigner: Send + Sync {
+    /// Returns the available accounts for this signer.
+    fn accounts(&self) -> Vec<Address>;
+
+    /// Returns `true` whether this signer can sign for this address
+    fn is_signer_for(&self, addr: Address) -> bool {
```
Comment (rkrasiuk) [inline crates/net/rpc/src/eth/signer.rs:16]: shall we accept addresses as ref? same for methods below 
```suggestion
    fn is_signer_for(&self, addr: &Address) -> bool {
```

**Comment by onbjerg** [comment]
Should we even support this namespace? This requires us to have a keystore impl + corresponding CLI plumbing. Some nodes skip this entirely since it is rarely used (I forget which)

If we should have a keystore we should create a tracking issue and break it into smaller tasks

**Comment by mattsse** [comment]
> Should we even support this namespace? 

signers are required for the `eth_` namespace, `eth_sendTransaction` `eth_sign` , `eth_signTypedData_v4`

so we definitely need something like this. abstracting this over an `EthSigner` trait is an easy solution imo.

**Comment by mattsse** [comment]
so also no support for `eth_sendTransaction`?

> so unless you specifically store keys side-by-side with your node

yeh this should be empty by default.
still, I think we want to support these.

**Comment by onbjerg** [comment]
>so also no support for eth_sendTransaction?

Yeah, wallets just use `eth_sendRawTransaction`

**Review by rkrasiuk — APPROVED** [review]
lgtm

**Review by gakonst — APPROVED** [review]
Good with me, these APIs are still used across the eco in weird setups, so I would still want us to support them. It's true that applications / wallets etc. use eth_sendRawTransaction (all/most).

**Pull request merged**